cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -149,6 +156,23 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	if opts.Private {
 		options = append(options, "private")
 	}
+	if opts.VerityHashDevice != "" && opts.VerityRootHash == "" {
+		return fmt.Errorf("cannot mount %q at %q: mount with dm-verity was requested but a root hash was not specified", what, where)
+	}
+	if opts.VerityRootHash != "" && opts.VerityHashDevice == "" {
+		return fmt.Errorf("cannot mount %q at %q: mount with dm-verity was requested but a hash device was not specified", what, where)
+	}
+	if opts.VerityHashOffset != 0 && (opts.VerityHashDevice == "" || opts.VerityRootHash == "") {
+		return fmt.Errorf("cannot mount %q at %q: mount with dm-verity was requested but a hash device and root hash were not specified", what, where)
+	}
+	if opts.VerityHashDevice != "" && opts.VerityRootHash != "" {
+		options = append(options, fmt.Sprintf("verity.roothash=%s", opts.VerityRootHash))
+		options = append(options, fmt.Sprintf("verity.hashdevice=%s", strings.ReplaceAll(opts.VerityHashDevice, ",", `\,`)))
```

Comment:
I think it is very important to have a clear picture of who is parsing this and how it is done. I've done by best to figure out how this is used and this is what I think happens:

- options are given to systemd-mount from systemd
- then options are given to libmount from util-linux in `libmount/src/hook_veritydev.c`
- then some if this uses cryptsetup to setup the device

As far as I can see, there's no escaping anywhere. So after the mount options are parsed by libmount, all paths are verbatim. I'm trying to see if libmount has some escape logic.